ziti-springboot-client/src/main/java/org/openziti/springboot/client/web/config/ZitiHttpClientConfiguration.java

```diff
@@ -42,126 +40,125 @@
+  private static final int DEFAULT_RESPONSE_TIMEOUT = 60 * 1000;
 
   // The default time to keep a connection alive.
   private static final long DEFAULT_KEEP_ALIVE_TIME_MILLIS = 20 * 1000;
 
   private ZitiConnectionSocketFactory zitiConnectionSocketFactory;
   private ZitiSSLConnectionSocketFactory zitiSSLConnectionSocketFactory;
 
-  @ConditionalOnMissingBean(name = "zitiRestTemplate")
-  @Bean("zitiRestTemplate")
-  public RestTemplate restTemplate(@Qualifier("zitiRestTemplateBuilder") RestTemplateBuilder restTemplateBuilder) {
+  @ConditionalOnProperty(value = "spring.ziti.client.rest-template.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public RestTemplate zitiRestTemplate(@Qualifier("zitiRestTemplateBuilder") RestTemplateBuilder restTemplateBuilder) {
     return restTemplateBuilder.build();
   }
 
-  @ConditionalOnMissingBean(name = "zitiRestTemplateBuilder")
-  @Bean("zitiRestTemplateBuilder")
-  public RestTemplateBuilder restTemplateBuilder(@Qualifier("zitiHttpClient") HttpClient httpClient) {
+  @ConditionalOnProperty(value = "spring.ziti.client.rest-template-builder.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public RestTemplateBuilder zitiRestTemplateBuilder(@Qualifier("zitiHttpClient") HttpClient httpClient) {
     return new RestTemplateBuilder()
-        .requestFactory(() -> clientHttpRequestFactory(httpClient));
+        .requestFactory(() -> zitiClientHttpRequestFactory(httpClient));
   }
 
-  @ConditionalOnMissingBean(name = "zitiClientHttpRequestFactory")
-  @Bean("zitiClientHttpRequestFactory")
-  public ClientHttpRequestFactory clientHttpRequestFactory(@Qualifier("zitiHttpClient") HttpClient httpClient) {
-    HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
+  @ConditionalOnProperty(value = "spring.ziti.client.request-factory.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public ClientHttpRequestFactory zitiClientHttpRequestFactory(@Qualifier("zitiHttpClient") HttpClient httpClient) {
+    final HttpComponentsClientHttpRequestFactory clientHttpRequestFactory = new HttpComponentsClientHttpRequestFactory();
     clientHttpRequestFactory.setHttpClient(httpClient);
     return clientHttpRequestFactory;
   }
 
-  @ConditionalOnMissingBean(ZitiConnectionSocketFactory.class)
+  @ConditionalOnProperty(value = "spring.ziti.client.context.enabled", havingValue = "true", matchIfMissing = true)
   @Bean
-  public ZitiConnectionSocketFactory zitiConnectionSocketFactory(
+  public ZitiContext zitiContext(
       @Value("${spring.ziti.httpclient.identity.file:}") Resource identityFile,
       @Value("${spring.ziti.httpclient.identity.password:}") String password) throws IOException {
 
-    if (zitiConnectionSocketFactory == null) {
+    if (identityFile == null) {
+      throw new BeanCreationException("To use the default ziti setup please add a property named "
```

Comment:
> To use the default ziti setup
?
Seems like this should be "No identity file specified." Also I prefer the previous message with the whole "Define one with `@Bean` or add a property..." Can one not use `@Bean` any longer?